pkg/landscaper/gardenlet/apis/imports/types.go

```diff
@@ -0,0 +1,96 @@
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imports
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	gardenletconfigv1alpha1 "github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// LandscaperGardenletImport defines the landscaper import for the Gardenlet.
+// structure defined in Blueprint
+type LandscaperGardenletImport struct {
+	metav1.TypeMeta
+	// RuntimeCluster is the landscaper target containing the kubeconfig for the cluster
+	// where the Gardenlet should be deployed.
+	// This is the Kubernetes cluster targeted as Seed (via in-cluster mounted service account token),
+	// if not otherwise specified in `.componentConfiguration.seedClientConnection.kubeconfig`.
+	RuntimeCluster Target
+	// GardenCluster is the landscaper target containing the kubeconfig for the
+	// Garden cluster (having Gardener resource groups!)
+	GardenCluster Target
+	// ImageVectorOverwrite contains the image vector override
+	ImageVectorOverwrite *string
+	// ImageVectorOverwrite contains the image vector override for components deployed by the gardenlet
+	ComponentImageVectorOverwrites *runtime.RawExtension
+	// SeedBackup contains configuration for an optional backup provider for the Seed cluster registered by the Gardenlet
+	// required when gardenlet.componentConfiguration.seedConfig.backup.secretRef is not set
+	// backup secret is deployed into the Garden cluster
+	SeedBackup *SeedBackup
+	// RevisionHistoryLimit is the revision history limit for the Gardenlet deployment
+	// Defaults to 10
+	RevisionHistoryLimit *int32
+	// Resources are the resource requirements for the Gardenlet pod
+	Resources *corev1.ResourceRequirements
+	// ComponentConfiguration is the Gardenlet component configuration
+	// NOTE: please set componentConfiguration.SeedConfig.Spec.Settings.VerticalPodAutoscaler.Enabled to enable VPA for the configured Seed
+	ComponentConfiguration gardenletconfigv1alpha1.GardenletConfiguration
+}
+
+// SeedBackup contains configuration for an optional backup provider for the Seed cluster registered by the Gardenlet
+type SeedBackup struct {
+	// Provider is the provider name {aws,gcp,...}
+	Provider string
+	// Credentials contains provider specific credentials
+	// Please check the documentation of the respective extension provider for the concrete format
+	Credentials *json.RawMessage
+}
+
+// Taken from github.com/gardener/landscaper/pkg/apis/core/v1alpha1
+// Avoid importing the Landscaper repository for a simple type
```

Comment:
Why? Please import it. If it pulls too many unrelated packages then the repository structure of the landscaper should be improved to isolate the types/constants into independent packages.

Reply:
@schrodit when importing the structures there are conflicting packages that break the Gardener build. 
Is it possible to reduce the packages that have to be imported when importing github.com/gardener/landscaper/pkg/apis/core/v1alpha1 ?



> github.com/docker/docker v1.4.2-0.20200203170920-46ec8731fbce/go.mod h1:eEKB0N0r5NX/I1kEveEz05bcu8tLC/8azJZsviup8Sk=
github.com/docker/docker-credential-helpers v0.6.3/go.mod h1:WRaJzqw3CTB9bk10avuGsjVBZsD05qeibJ1/TYlvc0Y=
github.com/docker/go-connections v0.4.0/go.mod h1:Gbd7IOopHjR8Iph03tsViu4nIes5XhDvyHbTtUxmeec=
github.com/docker/go-metrics v0.0.0-20180209012529-399ea8c73916/go.mod h1:/u0gXw0Gay3ceNrsHubL3BtdOL2fHf93USgMTe0W5dI=

>github.com/dnaeon/go-vcr v1.0.1/go.mod h1:aBB1+wY4s93YsC3HHjMBMrwTj2R9FHDzUr9KyGc8n1E=
github.com/docker/cli v0.0.0-20200130152716-5d0cf8839492/go.mod h1:JLrzqnKDaYBop7H2jaqPtU4hHvMKP+vjCwu2uszcLI8=
github.com/docker/distribution v0.0.0-20191216044856-a8371794149d/go.mod h1:0+TTO4EOBfRPhZXAeF1Vu+W3hHZ8eLp8PgKVZlcvtFY=
github.com/docker/distribution v2.7.1+incompatible/go.mod h1:J2gT2udsDAN96Uj4KfcMRqY0/ypR+oyYUYmja8H+y+w=
g

For one there are other versions of existing dependenices  (we would have to use replace command for most of them).
But also there are a bunch of  of additionional packages that I hope we can get rid of:

> github.com/daviddengcn/go-colortext v0.0.0-20160507010035-511bcaf42ccd/go.mod h1:dv4zxwHi5C/8AeI+4gX4dCWOIvNi7I6JCSX0HvlKPgE=
github.com/deislabs/oras v0.8.1/go.mod h1:Mx0rMSbBNaNfY9hjpccEnxkOqJL6KGjtxNHPLC4G4As=
github.com/denisenkom/go-mssqldb v0.0.0-20191001013358-cfbb681360f0/go.mod h1:xbL0rPBG9cCiLr28tMa8zpbdarY27NDyej4t/EjAShU=
github.com/denverdino/aliyungo v0.0.0-20190125010748-a747050bb1ba/go.mod h1:dV8lFg6daOBZbT6/BDGIz6Y3WFGn8juu6G+CQ6LHtl0=
github.com/dgrijalva/jwt-go v0.0.0-20170104182250-a601269ab70c/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=

> github.com/containerd/cgroups v0.0.0-20190919134610-bf292b21730f/go.mod h1:OApqhQ4XNSNC13gXIwDjhOQxjWa/NxkwZXJ1EvqT0ko=
github.com/containerd/console v0.0.0-20180822173158-c12b1e7919c1/go.mod h1:Tj/on1eG8kiEhd0+fhSDzsPAFESxzBBvdyEgyryXffw=
github.com/containerd/containerd v1.3.0-beta.2.0.20190828155532-0293cbd26c69/go.mod h1:bC6axHOhabU15QhwfG7w5PipXdVtMXFTttgp+kVtyUA=
github.com/containerd/containerd v1.3.2/go.mod h1:bC6axHOhabU15QhwfG7w5PipXdVtMXFTttgp+kVtyUA=
github.com/containerd/continuity v0.0.0-20190426062206-aaeac12a7ffc/go.mod h1:GL3xCUCBDV3CZiTSEKksMWbLE66hEyuu9qyDOOqM47Y=
github.com/containerd/continuity v0.0.0-20200107194136-26c1120b8d41/go.mod h1:Dq467ZllaHgAtVp4p1xUQWBrFXR9s/wyoTpG8zOJGkY=
github.com/containerd/fifo v0.0.0-20190226154929-a9fb20d87448/go.mod h1:ODA38xgv3Kuk8dQz2ZQXpnv/UZZUHUCL7pnLehbXgQI=
github.com/containerd/go-runc v0.0.0-20180907222934-5a6d9f37cfa3/go.mod h1:IV7qH3hrUgRmyYrtgEeGWJfWbgcHL9CSRruz2Vqcph0=
github.com/containerd/ttrpc v0.0.0-20190828154514-0e0f228740de/go.mod h1:PvCDdDGpgqzQIzDW1TphrGLssLDZp2GuS+X5DkEJB8o=
github.com/containerd/typeurl v0.0.0-20180627222232-a93fcdb778cd/go.mod h1:Cm3kwCdlkCfMSHURc+r6fwoGH6/F1hH3S4sg0rLFWPc=